premailer/tests/test_premailer.py

```diff
@@ -2984,3 +2984,57 @@ def test_pseudo_selectors_without_selector(self):
         p = Premailer(html, exclude_pseudoclasses=False, keep_style_tags=True)
         result_html = p.transform()
         compare_html(expect_html, result_html)
+
+    def test_preserve_handlebar_syntax_false(self):
+        """Demonstrate encoding of handlebar syntax with preservation disabled.
+
+        Original issue: https://github.com/peterbe/premailer/issues/248
+        """
+
+        html = """
+            <html>
+            <img src="{{ data | default: 'Test.' }}">
+            <a href="{{ data | default: "Test." }}"></a>
```

Comment:
```suggestion
            <a href="{{ data | default: "Test & Testing" }}"></a>
```

Reply:
Or, 
```suggestion
            <a href="{{ data | default: "Test & <code>" }}"></a>
```

Reply:
Because, that's valid Handlebar code, right?

Reply:
Good point. With this implementation, symbols like `&`, `<`, and `>` get encoded and `unquote()` doesn't successfully decode them back to their original symbols. I'm not quite sure where to begin trying to solve that. But, I'll be doing some research.

Reply:
A solution that I've come up with is using `html.unescape()` to replace the named and character references with their original unicode characters.
In addition to discovering `unescape()`, I also found `escape()` which I've decided to replace my usage of `replace()` that replaced `"` with `%22` (which I'll attempt to justify in the below comment).